src/components/input/template.test.js

```diff
@@ -0,0 +1,120 @@
+/* globals describe, it, expect */
+
+const { render, getExamples } = require('../../../lib/jest-helpers')
+
+const examples = getExamples('input')
+
+describe('Input', () => {
+  describe('by default', () => {
+    it('renders with classes', () => {
+      const { $ } = render('input', {
+        classes: 'app-c-input--custom-modifier'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.hasClass('app-c-input--custom-modifier')).toBeTruthy()
+    })
+
+    it('renders with id', () => {
+      const { $ } = render('input', {
+        id: 'my-input',
+        name: 'my-input-name'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('id')).toEqual('my-input')
+    })
+
+    it('renders with name', () => {
+      const { $ } = render('input', {
+        id: 'my-input',
+        name: 'my-input-name'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('name')).toEqual('my-input-name')
+    })
+
+    it('renders with type="text"', () => {
+      const { $ } = render('input')
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('type')).toEqual('text')
+    })
+
+    it('renders with value', () => {
+      const { $ } = render('input', {
+        id: 'my-input',
+        name: 'my-input-name',
+        value: 'QQ 12 34 56 C'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.val()).toEqual('QQ 12 34 56 C')
+    })
+
+    it('renders with attributes', () => {
+      const { $ } = render('input', {
+        id: 'my-input',
+        name: 'my-input-name',
+        attributes: {
+          'aria-describedby': 'my-label'
+        }
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('aria-describedby')).toEqual('my-label')
+    })
+
+    it('renders with label', () => {
+      const { $ } = render('input', {
+        id: 'my-input',
+        name: 'my-input-name',
+        label: {
+          'text': 'National Insurance number'
+        }
+      })
+
+      const $label = $('.govuk-c-label')
+      expect($label).toBeTruthy()
+      expect($label.attr('for')).toEqual('my-input')
+    })
+
+    it('renders with error message', () => {
+      const { $ } = render('input', {
+        errorMessage: {
+          'text': 'Error message'
+        }
+      })
+
+      const $component = $('.govuk-c-input')
+      const $errorMessage = $('.govuk-c-error-message')
+      expect($component.hasClass('govuk-c-input--error')).toBeTruthy()
+      expect($errorMessage).toBeTruthy()
+    })
+  })
+
+  describe('default example', () => {
+    it('renders with label', () => {
+      const { $ } = render('input', examples.default)
+      const $label = $('.govuk-c-label')
+      expect($label.text()).toContain('National Insurance number')
```

Comment:
Should we have a test that the label is associated by `id` / `for`?

Reply:
Oh, we do, but it's further up. What's the thinking being splitting these up into a separate context?

Reply:
From line 98 I tested the examples that we have defined in the `yaml` so somehow make sure we're not breaking them. Before that line are 'by default' tests that cover the arguments, logic and relationship.